pkg/apis/flow/v1alpha1/jobflow_types.go

```diff
@@ -44,6 +49,13 @@ type Flow struct {
 	Name string `json:"name"`
 	// +optional
 	DependsOn *DependsOn `json:"dependsOn,omitempty"`
+	// +optional
+	Patch *Patch `json:"patch,omitempty"`
+}
+
+type Patch struct {
+	// +optional
+	Spec *v1alpha1.JobSpec `json:"spec,omitempty"`
```

Comment:
Why do we need to store this field here? Can't we use the known information to get the original jobtemplate object? Or directly patch itself?

Reply:
In my usage scenario, I still need to Patch some information to replace the content of the JobTemplate. Avoid partial differences causing an explosion in the number of JobTemplates.
User case:
```yaml
apiVersion: flow.volcano.sh/v1alpha1
kind: JobTemplate
metadata:
  name: a
spec:
  tasks:
    - spec:
          containers:
            - image: tf2.16-cuda11-xxx:v1.0.1
              command:
                - sh
                - -c
                - ./script.py --dateset a.txt --mode 1
              imagePullPolicy: IfNotPresent
              name: nginx
              resources:
                requests:
                  cpu: "1"
          restartPolicy: OnFailure
---
apiVersion: flow.volcano.sh/v1alpha1
kind: JobFlow
metadata:
  name: test-flow1
  namespace: default
spec:
  jobRetainPolicy: delete
  flows:
    - name: a
---
apiVersion: flow.volcano.sh/v1alpha1
kind: JobFlow
metadata:
  name: test-flow2
  namespace: default
spec:
  jobRetainPolicy: delete
    - name: a
      patch:
          spec:
             tasks:
               - spec:
                     containers:
                        - image: tf2.16-cuda11-xxx:v1.0.2 # just different image
                           command:
                              - sh
                              - -c
                              - ./script.py --dateset b.txt --mode 2 # just different args
```

Reply:
sorry for this. I don't see what patch does. It seems to make the whole cr configuration more complicated.